pkg/certificateprovider/certificate_provider.go

```diff
@@ -12,5 +13,30 @@ type CertStoreConfig map[string]string
 // CertificateProvider is an interface that defines methods to be implemented by a each certificate provider
 type CertificateProvider interface {
 	// Returns an array of certificates based on certificate properties defined in attrib map
-	GetCertificatesContent(ctx context.Context, attrib map[string]string) ([]types.Certificate, error)
+	GetCertificates(ctx context.Context, attrib map[string]string) ([]*x509.Certificate, error)
```

Comment:
One of the future items is to enable Certificate providers in the cosign verifier path. It looks like cosign load a [ECDSA](https://github.com/deislabs/ratify/blob/main/plugins/verifier/cosign/cosign.go#L150) key, will this interface work with cosign?  
We want to keep this interface generic so it works for all verifiers that requires a validation cert. We haven't started exploring the integration with Cosign yet, appreciate your insight here :)

Reply:
Will look into this and get back w/ some info :)

Reply:
What I've found as a complete cosign newbie:

* We're not actually using this interface right now- was just trying to get things lined up consistently to prepare for implementing the factory in the future - and to try to use common interfaces vs the Key Vault-specific `types.Certificate` one here
* This won't work as-is with our cosign code that exists today - it expects a plain public key rather than a certificate. I'd be wary of doing "not certificate things" in the "certificate provider". Might need a new "public key provider", or change the name to something else in the future to cover both use cases?
* Our linked usage of cosign functions points to functions labeled "TODO: move this to an internal package" and only seem to be used to parse a key. I wonder if there are more suitable functions in the `sigstore/signature` package directly?

If cosign users have and typically uses ECDSA *certificates*, it seems easy enough to use. This test case loads in a cert w/ an ECDSA key, and then uses it to load up a sigstore verifier:

```go
func TestUseX509WithSigstore(t *testing.T) {
	certString := "-----BEGIN CERTIFICATE-----\nMIIB+TCCAZ+gAwIBAgIUNK/+de2dV3kCMVOGj+X6mdyO6A8wCgYIKoZIzj0EAwIw\nUjELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAldBMRAwDgYDVQQHDAdSZWRtb25kMREw\nDwYDVQQKDAhUZXN0Q2VydDERMA8GA1UEAwwIdGVzdGNlcnQwHhcNMjMwMjA4MTkz\nNjAwWhcNMjQwMjA4MTkzNjAwWjBSMQswCQYDVQQGEwJVUzELMAkGA1UECAwCV0Ex\nEDAOBgNVBAcMB1JlZG1vbmQxETAPBgNVBAoMCFRlc3RDZXJ0MREwDwYDVQQDDAh0\nZXN0Y2VydDBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IABDLn+GbbWXyXcVuQ/PFP\nmC1f8FiXK7IBCATvOZXxQLJnU8B55mo1BTZn8yBzT1TJA4XfQktNrZYHIWwJkeFh\njgSjUzBRMB0GA1UdDgQWBBQ4wBIh6ub86EFcRvPss6dEPPd7IDAfBgNVHSMEGDAW\ngBQ4wBIh6ub86EFcRvPss6dEPPd7IDAPBgNVHRMBAf8EBTADAQH/MAoGCCqGSM49\nBAMCA0gAMEUCIQCaEZDL399k2fnwGQCluc32Xk8Dugsucv+niWZCIBtA4AIgDYBZ\n2HVNJ/29vqxAHG4mzNXH4uBZj9DvJpmYbzUvlOk=\n-----END CERTIFICATE-----\n"
	c1 := []byte(certString)

	// this works for certificates, but wouldn't work for plain public keys
	certs, err := DecodeCertificates(c1)
	if err != nil {
		t.Fatalf(err.Error())
	}

	// I havent verified anything with this, but this creates an ECDSAVerifier w/o errors
	verifier, err := signature.LoadVerifier(certs[0].PublicKey, crypto.SHA256)
	if err != nil {
		t.Fatalf(err.Error())
	}

	fmt.Printf("verifier: %+v", verifier)
}
```

That said, if cosign typically uses public keys directly - this approach (and therefore involving x509) probably wouldn't make much sense

I've seen some cosign refactoring issues/PR's, particularly around making it work with authenticated registries. Would it make sense to land this feature and then include "how do we handle certs/keys in a consistent way?" in the scope of the design for that improvement?

Reply:
I think it's fair to say that our current cosign implementation with respect to handling public keys and interacting with registries is not optimal. I'd be in favor of a general overhaul of how cosign plugin and integration with ORAS store is done. This one is going to require a fair bit of research and work as far as I can tell...

Reply:
Thanks so much for looking into this. I think you make a good point, certificate providers should return certificates. I think we can go ahead with the interface update.
"TODO: move this to an internal package" this message is worrying, we probably will need to revisit the cosign verifier soon.